src/main/java/gov/usgs/wma/waterdata/groundwater/DiscreteGroundWaterRules.java

```diff
@@ -0,0 +1,87 @@
+package gov.usgs.wma.waterdata.groundwater;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import org.springframework.util.StringUtils;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
+/**
+ * Applies business rules to a domain object.
+ * It is assumed that this object can be modified.
+ */
+public class DiscreteGroundWaterRules {
+
+	//Threadsafe factory
+	private static final JsonFactory jsonFactory = new JsonFactory();
+
+	/**
+	 * Apply business rules to a DiscreteGroundWater, modifying it in place.
+	 * @param domObj
+	 */
+	public void apply(DiscreteGroundWater domObj) {
+
+		//Rule:  The AQ measurement reading qualifiers (multi-values) are mapped to a single NWISWeb lev_status_cd.
+		//If the AQ qualifier(s) are not recognized as valid AQ qualifiers, it is ignored.
+		//If the AQ qualifier(s) are empty or just not one that is mapped, it is assumed to be an unqualified measurement.
+		//Ref:  https://internal.cida.usgs.gov/jira/browse/IOW-667
+		//
+		//There is no order or preference among the multi-valued AQ qualifiers to select a single one for NWISWeb.
+		//To be repeatable, it is assumed that the qualifiers will be matched in the order of the LevelStatusCode enum,
+		//not in order of the values in the db column (order is not repeatable within a JSONB object).
+		{
+			String orgQualStr = StringUtils.trimWhitespace(domObj.readingQualifiers);
+			String newQualStr = "";   //acceptable default value if no other found
+
+			if (! StringUtils.isEmpty(orgQualStr)) {
+				try {
+
+					List<String> aqQuals = new ArrayList();
+
+					JsonParser parser = jsonFactory.createParser(orgQualStr);
+
+					while (! parser.isClosed()) {
+
+						JsonToken token = parser.nextToken();
+
+						if (token != null && token.isScalarValue()) {
+							String val = StringUtils.trimWhitespace(parser.getValueAsString());
+
+							if (!StringUtils.isEmpty(val)) {
+								aqQuals.add(StringUtils.trimWhitespace(parser.getValueAsString()));
```

Comment:
I know that tests would be passing but I have a question about token vs parser. It seems odd that we get the value off the parser when we just obtained a current token.

Reply:
The syntax of the Jackson parser is confusing.  JsonToken is just an enum, intended to tell you what what the next 'thing' is.  nextToken() advances the pointer and gives you the token, but to actually get the content at the pointer, 'get' needs to be called on the parser.

Reply:
oh, that is a strange concept. no, actually I think it is a misnamed method. not that you should change the lib, but that method should be something like currentTokenType()